### PR TITLE
allow running with masked string instead of mask

### LINKED ISSUE #3: add masked commandline option
Currently, only a mask file can be used for submission.
Alternative option could be to allow a masked commandline input like `--commandparam1=%(param1)s  --commandparam2=%(param2)s` .

### PR DESCRIPTION
**Pull request by mihofer**
Fixes #3 and documentation

### CODE REVIEW

File: pylhc_submitter/htc/utils.py [hunk 1]
```diff
@@ -186,7 +187,16 @@ def write_bash(
                 f.write(f"{SHEBANG}\n") 
             if output_dir is not None:
                 f.write(f"mkdir {str(output_dir)}\n")
-            f.write(f"{exec_path}{str(job_dir / job[COLUMN_JOB_FILE])}{cmds}\n")
+            f.write(f"{exec_path}")
+
+            if is_mask_file(mask):
+                f.write(f"{str(job_dir / job[COLUMN_JOB_FILE])}")
+            elif is_mask_string(mask):
```
Comment (JoschD) [inline pylhc_submitter/htc/utils.py:194]: Shouldn't this just be `else` ? And then you don't need the second function at all...

File: pylhc_submitter/htc/utils.py [hunk 2]
```diff
@@ -236,6 +246,17 @@ def _maybe_put_in_quotes(key, value):
     return value
 
 
+def is_mask_file(mask):
+    is_file=False
+    try:
+        is_file = Path(mask).is_file()
+    except OSError:
+        pass
+    return is_file
```
Comment (JoschD) [inline pylhc_submitter/htc/utils.py:255]:
```python
def is_mask_file(mask):    
    try:
        return Path(mask).is_file()    
    except OSError:
        return False
```
Shorter and nicer

File: pylhc_submitter/htc/utils.py [hunk 3]
```diff
@@ -186,7 +187,16 @@ def write_bash(
...
-            f.write(f"{exec_path}{str(job_dir / job[COLUMN_JOB_FILE])}{cmds}\n")
+            f.write(f"{exec_path}")
+
+            if is_mask_file(mask):
+                f.write(f"{str(job_dir / job[COLUMN_JOB_FILE])}")
+            else:
+                replace_columns = [column for column in job.index.tolist() if column not in [COLUMN_SHELL_SCRIPT, COLUMN_JOB_DIRECTORY, COLUMN_JOB_FILE]]
+                f.write(mask % dict(zip( replace_columns, job[replace_columns])))
+
+            f.write(f"{cmds}")
```
Comment (JoschD) [inline pylhc_submitter/htc/utils.py:198]: unnecessary f-string

File: pylhc_submitter/htc/utils.py [hunk 4]
```diff
@@ -186,7 +187,16 @@ def write_bash(
                 f.write(f"{SHEBANG}\n") 
             if output_dir is not None:
                 f.write(f"mkdir {str(output_dir)}\n")
-            f.write(f"{exec_path}{str(job_dir / job[COLUMN_JOB_FILE])}{cmds}\n")
+            f.write(f"{exec_path}")
```
Comment (JoschD) [inline pylhc_submitter/htc/utils.py:190]: unnecessary f-string

File: pylhc_submitter/htc/utils.py [hunk 5]
```diff
@@ -186,7 +187,16 @@ def write_bash(
                 f.write(f"{SHEBANG}\n") 
             if output_dir is not None:
                 f.write(f"mkdir {str(output_dir)}\n")
-            f.write(f"{exec_path}{str(job_dir / job[COLUMN_JOB_FILE])}{cmds}\n")
+            f.write(f"{exec_path}")
+
+            if is_mask_file(mask):
+                f.write(f"{str(job_dir / job[COLUMN_JOB_FILE])}")
```
Comment (JoschD) [inline pylhc_submitter/htc/utils.py:193]: unnecessary f-string

File: pylhc_submitter/htc/utils.py [hunk 6]
```diff
@@ -186,7 +187,16 @@ def write_bash(
...
             if output_dir is not None:
                 f.write(f"mkdir {str(output_dir)}\n")
-            f.write(f"{exec_path}{str(job_dir / job[COLUMN_JOB_FILE])}{cmds}\n")
+            f.write(f"{exec_path}")
+
+            if is_mask_file(mask):
+                f.write(f"{str(job_dir / job[COLUMN_JOB_FILE])}")
+            else:
+                replace_columns = [column for column in job.index.tolist() if column not in [COLUMN_SHELL_SCRIPT, COLUMN_JOB_DIRECTORY, COLUMN_JOB_FILE]]
+                f.write(mask % dict(zip( replace_columns, job[replace_columns])))
```
Comment (JoschD) [inline pylhc_submitter/htc/utils.py:196]: unnecessary space

**Review by JoschD — CHANGES_REQUESTED** [review]

**Review by JoschD — APPROVED** [review]

**Review by fsoubelet — DISMISSED** [review]
Good for me too. We should probably have an issue for dedicated examples in the documentation.

**Comment by mihofer** [comment]
before we go, bump to 1.0.0?
means also changelog should be fixed ..

**Comment by fsoubelet** [comment]
> before we go, bump to 1.0.0?

I would say no as I'm attached to `SemVer`.

**Review by fsoubelet — APPROVED** [review]

**Pull request merged**